packages/cozy-doctypes/src/File.js

```diff
@@ -148,6 +148,64 @@ class CozyFile extends Document {
       return `${filenameWithoutExtension}_1`
     }
   }
+
+  /**
+   * The goal of this method is to upload a file based on a conflict strategy.
+   * Be careful: We need to check if the file exists by doing a statByPath query
+   * before trying to upload the file since if we post and the stack return a
+   * 409 conflict, we will get a SPDY_ERROR_PROTOCOL on Chrome. This is the only
+   * viable workaround
+   * If there is no conflict, then we upload the file.
+   * If there is a conflict, then we applied the conflict strategy : `erase` or `rename`
+   * @param {String} name File Name
+   * @param {ArrayBuffer} file data
+   * @param {String} dirId dir id where to upload
+   * @param {String} onConflict Actually only 2 hardcoded strategies 'erase' or 'rename'
+   */
+  static async uploadFileWithConflictStrategy(name, file, dirId, onConflict) {
```

Comment:
nit: I think `conflictStrategy` would be a better name for `onConflict`. It may be because of react, but when I read `onSomething`, I'm waiting for it to be a function.